### PR TITLE
GP11 : TEE_OpenPersistentObject panic reason

### PR DESCRIPTION
**Pull request by cedric-chaumont-st-dev**
Signed-off-by: Cedric Chaumont <cedric.chaumont@st.com>

### CODE REVIEW

File: lib/libutee/tee_api_objects.c [hunk 1]
```diff
@@ -397,20 +397,43 @@ TEE_Result TEE_OpenPersistentObject(uint32_t storageID, void *objectID,
...
+		goto err;
+	}
+
+	res = utee_storage_obj_open(storageID, objectID, objectIDLen, flags,
 				     object);
+
+	if (res == TEE_SUCCESS)
+		goto out;
+err:
+	if (res == TEE_ERROR_ITEM_NOT_FOUND ||
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_objects.c]: If you added `res == TEE_SUCCESS` here you could get rid of the `if` just above.

File: lib/libutee/tee_api_objects.c [hunk 2]
```diff
@@ -397,20 +397,43 @@ TEE_Result TEE_OpenPersistentObject(uint32_t storageID, void *objectID,
...
 				    TEE_ObjectHandle *object)
 {
-	if (storageID != TEE_STORAGE_PRIVATE)
-		return TEE_ERROR_ITEM_NOT_FOUND;
+	TEE_Result res;
 
-	if (objectID == NULL)
-		return TEE_ERROR_ITEM_NOT_FOUND;
+	if (storageID != TEE_STORAGE_PRIVATE) {
+		res = TEE_ERROR_ITEM_NOT_FOUND;
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_objects.c]: Why not return this directly, there's nothing special for this at the end any way.

Comment (pascal-brand-st-dev) [inline lib/libutee/tee_api_objects.c]: I agree the way it is currently written is redundant.

But it clearly shows the GP API, with accepted return codes and panic case, that are not spread everywhere in the code, but located only at the end of the function.  So I am quite in favor of this approach.

We may need a referee  :)  ?

**Comment by cedric-chaumont-st-dev** [comment]
Rebased.

File: lib/libutee/tee_api_objects.c (hunk 2 above)
Comment (jenswi-linaro) [inline lib/libutee/tee_api_objects.c]: Either way it OK, but I wouldn't have written it like this myself.

**Comment by cedric-chaumont-st-dev** [comment]
Rebased.

File: lib/libutee/tee_api_objects.c (hunk 1 above)
Comment (pascal-brand-st-dev) [inline lib/libutee/tee_api_objects.c]: +1.
`out:` label can also be removed.

The same applies in other patches I think.

**Comment by cedric-chaumont-st-dev** [comment]
Updated.

**Comment by pascal-brand-st-dev** [comment]
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

**Comment by cedric-chaumont-st-dev** [comment]
Squashed. Rebased.

**Comment by jenswi-linaro** [comment]
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

**Comment by cedric-chaumont-st-dev** [comment]
Updated. Tested.

**Pull request merged**